moochub-schema.json

```diff
@@ -1,555 +1,3105 @@
+                      "example": "This is an image. It shows ..."
+                    },
+                    "type": {
+                      "type": "string",
+                      "description": "The type of the object.",
+                      "enum": [
+                        "ImageObject"
+                      ]
+                    },
+                    "contentUrl": {
+                      "type": "string",
+                      "format": "iri",
+                      "description": "The IRI pointing at the actual image.",
+                      "example": "https://openhpi.azureedge.net/assets/defaults/course-bb5a40d888217b3563ca4da5d868e770f43bb9ed70f1619147af98162b86ec3e.png"
+                    },
+                    "license": {
+                      "type": "array",
+                      "description": "A list of licenses for the Image.",
+                      "uniqueItems": true,
+                      "minItems": 1,
+                      "items": {
+                        "title": "License",
+                        "type": "object",
+                        "description": "This schema describes the attributes of a license.\n",
+                        "properties": {
+                          "identifier": {
+                            "description": "A license according to https://spdx.org/licenses/ or \"proprietary\".",
+                            "example": "https://creativecommons.org/publicdomain/zero/1.0",
+                            "format": "iri",
+                            "oneOf": [
+                              {
+                                "format": "iri",
+                                "example": "https://creativecommons.org/publicdomain/zero/1.0",
+                                "description": " A generic, general license according to https://spdx.org/licenses/."
+                              },
+                              {
+                                "enum": [
+                                  "proprietary"
+                                ],
+                                "example": "proprietary",
+                                "description": "Marking the course as proprietary."
+                              }
+                            ],
+                            "type": "string"
+                          },
+                          "url": {
+                            "type": [
+                              "string",
+                              "null"
+                            ],
+                            "format": "iri",
+                            "example": null,
+                            "description": "A link to the specific license document (if any, otherwise set to \"null\" - the identifier pointing at the general license is to be considered the license document then). Has to be \"null\" if identifier is \"proprietary\"."
```

Comment:
I don't really like tri-state / implicit states.

Reply:
What do you mean by this? Can you explain it a little more?

Reply:
I think I got what you mean now... but I do not see a better solution. If something has a license it shall not be listed as proprietary and if it is proprietary it should not have any license depicted... any ideas?